### DISCUSSION TITLE
Simplify startup guide: Running Gitleaks with Docker Without Installing Python or Pre-commit

### DISCUSSION THREAD

**Discussion by mulder999** [zricethezav/gitleaks#1666] [General]
No need to install Python, pip, or pre-commit.

If Docker is set up, the only thing you need to do is to set this in file `.git/hooks/pre-commit` (Windows variant): 

```(bash)
#!/bin/bash

MSYS_NO_PATHCONV=1 docker run --rm --network=none -v $(pwd):/repo:ro -w /repo ghcr.io/gitleaks/gitleaks:latest git --pre-commit --redact --staged --verbose
```

For *nix systems, you can omit the `MSYS_NO_PATHCONV=1` part.

**Comment by thedanielmark** [comment]
@mulder999 Thanks for this! Saved me after hours of trying to debug other methods.
